### PR TITLE
Resources API endpoints documentation

### PR DESCRIPTION
**Pull request by Aquassaut**

### CODE REVIEW

**Comment by vbarbaresi** [comment]
I already brought this up a few months ago, but do you want to document all URLs with a trailing slash? I thought we agreed that `/api/management/v2/resource/{RESOURCE_UID}` looks better than `/api/management/v2/resource/{RESOURCE_UID}/`

The ending slash could mean "all" resources, like `GET /api/management/v2/resources/` could be acceptable
See https://softwareengineering.stackexchange.com/questions/186959/trailing-slash-in-restful-api for opinions

**Comment by Aquassaut** [comment]
I have no opinion on that, so I agree by default. I'll make the necessary modifications to the doc if everybody agrees too

File: source/includes/_dataset_resources.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
+# Dataset Resources
+
+## The resource object
+
+The resource object describe a resource on the OpenDataSoft platform. It is composed of a URL, a title, a type, a parameter object and a credentials object. The URL is where data will be pulled to populate the dataset. Resources urls can (and often do!) point to [files](#files) uploaded to the platform using the `odsfile://` url scheme.
```
Comment (juliendbg) [inline source/includes/_dataset_resources.md]: The resource object _describes_ a resource

File: source/includes/_dataset_resources.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+`credentials` <br> *object* | resource credentials
+
+## List dataset resources
+
+This endpoint is meant to list all resources that are linked to a dataset.
+
+> Definition
+
+```HTTP
+GET https://{YOURDOMAIN}.opendatasoft.com/api/management/v2/resources/
```
Comment (juliendbg) [inline source/includes/_dataset_resources.md]: Isn't that supposed to be `/api/management/v2/{dataset_uid}/resources/`? (same for all examples below)

File: source/includes/_dataset_resources.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    "type": "csvfile",
+    "params": {
+        "headers_first_row": false,
+        "separator": ";"
+    },
+    "credentials": {}
+}
+```
+
+A new resource is created using the resource object sent in the body, and echos back the object, with its newly generated resource_uid on success.
```
Comment (juliendbg) [inline source/includes/_dataset_resources.md]: and _echoes_ back

File: source/includes/_dataset_resources.md (hunk 1 above)
Comment (Aquassaut) [reply source/includes/_dataset_resources.md]: :+1:

File: source/includes/_dataset_resources.md (hunk 2 above)
Comment (Aquassaut) [reply source/includes/_dataset_resources.md]: :+1:

File: source/includes/_dataset_resources.md (hunk 3 above)
Comment (Aquassaut) [reply source/includes/_dataset_resources.md]: :+1:

**Pull request merged**